x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/edit/index.tsx

```diff
@@ -205,19 +223,20 @@ const EditRulePageComponent: FC = () => {
         content: (
           <>
             <EuiSpacer />
-            <StepPanel loading={loading} title={ruleI18n.DEFINITION}>
-              {defineStep.data != null && (
+            <StepPanel loading={loading || isSavedQueryLoading} title={ruleI18n.DEFINITION}>
+              {defineStepDataWithSavedQuery != null && !isSavedQueryLoading && (
```

Comment:
Interesting that `defineStepDataWithSavedQuery` is either a specific object or undefined but there is a non strict check for `null`

<img width="613" alt="image" src="https://user-images.githubusercontent.com/3775283/194697467-c659ba9a-5ef7-4dd3-bbc5-752ca796a54d.png">

Reply:
non strict check for `null` eliminates `undefined`, so the value of `defineStepDataWithSavedQuery` will be object of `DefineStepRule` type.

Fairly common pattern across Kibana codebase (example from previous implementation):

```typescript
            <StepPanel loading={loading} title={ruleI18n.DEFINITION}>
              {defineStep.data != null && (
```